src/centos/scripts/start-marklogic.sh

```diff
@@ -165,31 +208,45 @@ function restart_check {
 #   $4 :  Option to return error or response code in case of error   
 ################################################################
 function curl_retry_validate {
-    local retry_count
-    local return_error="${4:-true}"
+    local retry_count response_code request
+    local return_error="${4:-true}" curl_options=$3 endpoint=$1
     for ((retry_count = 0; retry_count < N_RETRY; retry_count = retry_count + 1)); do
-        request="curl -m 30 -s -w '%{http_code}' $3 $1"
+        request="curl -m 30 -s -w '%{http_code}' $curl_options $endpoint"
         response_code=$(eval "${request}")
         if [[ ${response_code} -eq $2 ]]; then
             return "${response_code}"
         fi
         sleep ${RETRY_INTERVAL}
     done
     if [[ "${return_error}" = "false" ]] ; then
-        return "${response_code}"  
+        return "${response_code}"
     fi
     error "Expected response code ${2}, got ${response_code} from ${1}." exit
 }
 
+################################################################
+# Checks protocol for a host
+# input:  $1:     host name
+################################################################
+function get_host_protocol {
+    local hostname=$1 port="${2:-8001}" protocol="http" resp
```

Comment:
each local assignment statement should be in it's own line:
local hostname=$1
localport="${2:-8001}"
local https_error_message="You have attempted to access an HTTPS server using HTTP."
local resp=$(curl -s http://"${hostname}":"${port}")

Reply:
updated var declaration and assignment for variables